### ISSUE TITLE
x/build/dashboard: add js-wasm-node18 builder with Node.js 18 LTS

### ISSUE THREAD

**Issue by johanbrandhorst** [golang/go#57017]
Our current wasm NodeJS support policy is to support the latest LTS only [^1][^2]. The latest long term support release of NodeJS is [18](https://github.com/nodejs/release#release-schedule). The current js-wasm builders are [using version 14](https://github.com/golang/build/tree/master/env/js-wasm). We should update our builders to use the latest LTS so that we can evolve our use of NodeJS primitives and ensure we support users using the latest version of NodeJS.

[^1]: https://github.com/golang/go/issues/28360#issuecomment-474621552
[^2]: https://github.com/golang/go/issues/56860#issuecomment-1321273696

**Comment by johanbrandhorst** [comment]
CC @neelance

**Comment by johanbrandhorst** [comment]
I started looking into this and ran into some issues doing a simple upgrade to 18. I'm documenting it here for myself and others attempting this upgrade. Doing a simple replace of setup_14.x with [setup_18.x](https://deb.nodesource.com/setup_18.x) results in an error upgrading `gnupg`:
```
## Installing the NodeSource Node.js 18.x repo...


## Populating apt-get cache...

+ apt-get update
Get:1 http://deb.debian.org/debian sid InRelease [161 kB]
Get:2 http://deb.debian.org/debian sid/main amd64 Packages [9398 kB]
Fetched 9559 kB in 3s (3332 kB/s)   
Reading package lists... Done

## Installing packages required for setup: lsb-release gnupg...

+ apt-get install -y lsb-release gnupg
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libc6-dev : Breaks: libgcc-9-dev (< 9.3.0-5~) but 9.2.1-19 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
Error executing command, exiting
```

It seems to have something to do with a [debian upgrade](https://unix.stackexchange.com/questions/592657/full-upgrade-to-debian-testing-fails-due-to-libc6-dev-breaks-libgcc-8-dev), which leads me to believe that we may need to update the base image debian version, but this is all the debugging I've done so far.

**Comment by johanbrandhorst** [comment]
Changing from `sid` to `bullseye` fixes the error. I've submitted a CL with a fix for this: https://go-review.googlesource.com/c/build/+/455375.

**Comment by dmitshur** [comment]
I ended up not seeing this issue and mailed https://go.dev/cl/460036 while I was picking up some builder maintenance work this week (CC @golang/release). That CL takes a more gradual approach to upgrade the version by adding a new builder (with a different name), get it working, then drop the old one, instead of attempting to do an in-place upgrade. I think that should work better.

@johanbrandhorst I ran into the problem you described in https://github.com/golang/go/issues/57017#issuecomment-1333148796 too. I think we can keep sid, the problem is just that the "latest" tag is 3 years old because our newer [Cloud Build configuration](https://github.com/golang/build/blob/85db05fffd8387caea18f3d6f09d250b204482d7/env/linux-x86-sid/cloudbuild.yaml#L9-L11) targets the "cloudbuild" tag instead. In CL 460036 I switched to using the "cloudbuild", though I see now we should likely update "latest" instead.
